core/src/test/java/org/apache/brooklyn/util/core/osgi/OsgisTest.java

```diff
@@ -0,0 +1,117 @@
+
+public class OsgisTest {
+
+    private Framework framework;
+    private BundleContext bundleContext;
+    private Bundle myBundle1_0_0;
+    private Bundle myBundle1_1_0;
+    private Bundle otherBundle1_0_0;
+    private Bundle snapshotBundle1_0_0_snapshot;
+    
+    @BeforeMethod(alwaysRun=true)
+    public void setUp() throws Exception {
+        myBundle1_0_0 = Mockito.mock(Bundle.class);
+        Mockito.when(myBundle1_0_0.getSymbolicName()).thenReturn("mybundle");
+        Mockito.when(myBundle1_0_0.getVersion()).thenReturn(Version.parseVersion("1.0.0"));
+        
+        myBundle1_1_0 = Mockito.mock(Bundle.class);
+        Mockito.when(myBundle1_1_0.getSymbolicName()).thenReturn("mybundle");
+        Mockito.when(myBundle1_1_0.getVersion()).thenReturn(Version.parseVersion("1.1.0"));
+        
+        otherBundle1_0_0 = Mockito.mock(Bundle.class);
+        Mockito.when(otherBundle1_0_0.getSymbolicName()).thenReturn("otherbundle");
+        Mockito.when(otherBundle1_0_0.getVersion()).thenReturn(Version.parseVersion("1.0.0"));
+        
+        snapshotBundle1_0_0_snapshot = Mockito.mock(Bundle.class);
+        Mockito.when(snapshotBundle1_0_0_snapshot.getSymbolicName()).thenReturn("snapshotbundle");
+        Mockito.when(snapshotBundle1_0_0_snapshot.getVersion()).thenReturn(Version.parseVersion("1.0.0.SNAPSHOT"));
+        
+        bundleContext = Mockito.mock(BundleContext.class);
+        Mockito.when(bundleContext.getBundles()).thenReturn(new Bundle[] {myBundle1_0_0, myBundle1_1_0, otherBundle1_0_0, snapshotBundle1_0_0_snapshot});
+        
+        framework = Mockito.mock(Framework.class);
+        Mockito.when(framework.getBundleContext()).thenReturn(bundleContext);
+        
+    }
+    
+    @Test
+    public void testFindByNameAndVersion() throws Exception {
+        Bundle result = Osgis.bundleFinder(framework)
+                .symbolicName("mybundle")
+                .version("1.0.0")
+                .find()
+                .get();
+        assertEquals(result, myBundle1_0_0);
+    }
+
+    @Test
+    public void testFindAllByNameAndVersion() throws Exception {
+        List<Bundle> result = Osgis.bundleFinder(framework)
+                .symbolicName("mybundle")
+                .version("1.0.0")
+                .findAll();
+        assertEquals(result, ImmutableList.of(myBundle1_0_0));
+    }
+    
+    @Test
+    public void testFindByNameAndVersionRange() throws Exception {
+        List<Bundle> result = Osgis.bundleFinder(framework)
+                .symbolicName("mybundle")
+                .version("[1.0.0, 1.1.0)")
+                .findAll();
+        assertEquals(result, ImmutableList.of(myBundle1_0_0));
+        
+        List<Bundle> result2 = Osgis.bundleFinder(framework)
+                .symbolicName("mybundle")
+                .version("[1.0.0, 2.0.0)")
+                .findAll();
+        assertEquals(result2, ImmutableList.of(myBundle1_0_0, myBundle1_1_0));
```

Comment:
I think its worth adding the versions `2.0.0.SNAPSHOT` & `2.0.0` and making assertions around this boundary

Alos add checks for the range `[1.0.0, 2.0.0]`